pkg/issuer/acme/dns/dns.go

```diff
@@ -253,15 +253,22 @@ func (s *Solver) solverForChallenge(ctx context.Context, issuer v1alpha2.Generic
 		}
 	case providerConfig.Cloudflare != nil:
 		dbg.Info("preparing to create Cloudflare provider")
-		apiKeySecret, err := s.secretLister.Secrets(resourceNamespace).Get(providerConfig.Cloudflare.APIKey.Name)
-		if err != nil {
-			return nil, nil, fmt.Errorf("error getting cloudflare service account: %s", err)
+		apiKeySecret, keyErr := s.secretLister.Secrets(resourceNamespace).Get(providerConfig.Cloudflare.APIKey.Name)
+		apiTokenSecret, tokenErr := s.secretLister.Secrets(resourceNamespace).Get(providerConfig.Cloudflare.APIToken.Name)
+		if keyErr != nil && tokenErr != nil {
+			return nil, nil, fmt.Errorf("error getting cloudflare API key or API token: %s %s", keyErr, tokenErr)
 		}
 
 		email := providerConfig.Cloudflare.Email
-		apiKey := string(apiKeySecret.Data[providerConfig.Cloudflare.APIKey.Key])
+		var apiKey, apiToken string
+		if apiKeySecret != nil {
+			apiKey = string(apiKeySecret.Data[providerConfig.Cloudflare.APIKey.Key])
```

Comment:
Probably worth checking the bool value to ensure an expected value at that key has been set.